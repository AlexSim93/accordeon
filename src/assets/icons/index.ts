export { default as Chevron } from './chevron.svg';
