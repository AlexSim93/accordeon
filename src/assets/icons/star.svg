<svg width="24" height="24" viewBox="0 0 24 24" fill="none"
    xmlns="http://www.w3.org/2000/svg">
    <path d="M23.2269 8.50611H14.3574L11.6177 0L8.86955 8.50611L0 8.49748L7.18297 13.7599L4.43478 22.2574L11.6177 17.0036L18.7921 22.2574L16.0525 13.7599L23.2269 8.50611Z" fill="#00B67A"/>
    <path d="M16.6689 15.6838L16.0525 13.76L11.6177 17.0037L16.6689 15.6838Z" fill="#005128"/>
</svg>
