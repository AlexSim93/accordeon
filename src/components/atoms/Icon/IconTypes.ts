export type SizeType = 20 | 24 | 32 | 40 | 48 | 56 | 64;

export const ICON_SIZE: Record<
  'xs' | 's' | 'm' | 'ml' | 'l' | 'xl' | 'xxl',
  SizeType
> = {
  xs: 20,
  s: 24,
  m: 32,
  ml: 40,
  l: 48,
  xl: 56,
  xxl: 64,
};
