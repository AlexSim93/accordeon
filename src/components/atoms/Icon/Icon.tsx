import * as React from 'react';
import styled from 'styled-components';
import SVG from 'react-inlinesvg';
// eslint-disable-next-line no-unused-vars
import { ICON_SIZE, SizeType } from './IconTypes';

export const ICON_CLASSNAME = 'iconSvg';
export const WRAP_CLASSNAME = 'wrapSvg';

type Props = {
    type: string;
    fill?: string;
    size?: SizeType;
  };

type ComponentProps = {
    className: typeof ICON_CLASSNAME | typeof WRAP_CLASSNAME;
    size: SizeType;
  };

const IconComponent = styled.span<ComponentProps>`
  display: inline-block;
`;

const replaceFillInSVG = (code: string, fill: string) => (fill !== 'none'
  ? code
    .replace(/fill="[^none].*?"/g, `fill="${fill}"`)
    .replace(/stroke=".*?"/g, `stroke="${fill}"`)
  : code);

export const Icon = ({
  type,
  fill = 'none',
  size = ICON_SIZE.xs,
}: React.PropsWithChildren<Props>) => {
  const correctSize = Object.values(ICON_SIZE).includes(size)
    ? size
    : ICON_SIZE.xs;

  return (
    <IconComponent size={correctSize} className={WRAP_CLASSNAME}>
      {type === 'none' ? (
        <IconComponent size={correctSize} className={ICON_CLASSNAME} />
      ) : (
        <SVG
          src={type}
          className={ICON_CLASSNAME}
          height={correctSize}
          width={correctSize}
          preProcessor={(code) => replaceFillInSVG(code, fill)}
          cacheRequests
        />
      )}
    </IconComponent>
  );
};

Icon.displayName = 'Icon';
