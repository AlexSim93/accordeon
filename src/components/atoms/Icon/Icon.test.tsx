import * as React from 'react';
import renderer from 'react-test-renderer';
import { Icon } from './Icon';
import * as SvgIconImport from '../../../assets/icons';
import { ICON_SIZE } from './IconTypes';
import 'jest-styled-components';

describe('Icon renders correctly', () => {
  test('Icon renders correctly with type=Chevron fill=black and size="m"', () => {
    const tree = renderer
      .create(<Icon type={SvgIconImport.Chevron} fill="black" size={ICON_SIZE.m} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
