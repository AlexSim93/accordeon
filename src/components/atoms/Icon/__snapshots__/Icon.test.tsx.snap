// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly Icon renders correctly with type=Chevron fill=black and size="m" 1`] = `null`;
